### PR TITLE
Issue 8909 markdownlint

### PR DESCRIPTION
**Pull request by pbludov**
Issue #8909

Almost no changes in the existing .md files.

The one significant change is that now we have a top level header in the each file:
* [md041](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md041---first-line-in-file-should-be-a-top-level-header)

~~I vote to suppress this rule.~~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# Checkstyle
```
Comment (pbludov) [inline README.md:1]: I don't like it:

image

Now we have double title - one in the text, another in the image

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-PLEASE READ before removing!!
+# PLEASE READ before removing
```
Comment (pbludov) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: Here we are loosing `!!` due to [md041](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md041---first-line-in-file-should-be-a-top-level-header)

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 3]
```diff
@@ -1,3 +1,5 @@
+# Feature request template
```
Comment (pbludov) [inline .github/ISSUE_TEMPLATE/feature_request.md:1]: Again, [md041](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md041---first-line-in-file-should-be-a-top-level-header)

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -1,3 +1,5 @@
+# Bug report template
```
Comment (pbludov) [inline .github/ISSUE_TEMPLATE/bug_report.md:1]: Again, [md041](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md041---first-line-in-file-should-be-a-top-level-header)

File: README.md (hunk 1 above)
Comment (rnveach) [reply README.md:1]: Can you do something like:
````
# ![](https://raw.githubusercontent.com/checkstyle/resources/master/img/checkstyle-logos/checkstyle-logo-260x99.png)
````

Comment (pbludov) [reply README.md]: It works:

image

much better now

**Comment by rnveach** [comment]
There is a file conflict.

**Review by rnveach — APPROVED** [review]

**Comment by pbludov** [comment]
> There is a file conflict.

rebased

**Review by romani — APPROVED** [review]
Let's wait for CI.
Awesome update.

**Pull request merged**